### PR TITLE
Default to the renderer's resolution/multisample in `generateTexture`

### PR DESCRIPTION
**Pull request by dev7355608**
##### Description of change

`Extract` uses the renderer's resolution and multisample. I think `extract.pixels(generateTexture(object))` and `extract.pixels(object)` should give the same result.

`IGenerateTextureOptions.resolution` already made the claim that it defaults to the renderer's resolution, but that wasn't the case (`PIXI.settings.RESOLUTION` was the default):
>   The resolution / device pixel ratio of the texture being generated. Optional defaults to Renderer resolution.

##### Pre-Merge Checklist

- [ ] Tests and/or benchmarks are included
- [x] Documentation is changed or added
- [x] Lint process passed (`npm run lint`)
- [x] Tests passed (`npm run test`)

### CODE REVIEW

File: packages/core/src/renderTexture/GenerateTextureSystem.ts [hunk 1]
```diff
@@ -2,27 +2,26 @@ import { extensions, ExtensionType } from '@pixi/extensions';
...
+import type { IBaseTextureOptions } from '../textures/BaseTexture';
 
 const tempTransform = new Transform();
 
 // TODO could this just be part of extract?
-export interface IGenerateTextureOptions
+export interface IGenerateTextureOptions extends IBaseTextureOptions
 {
-    /** The scale mode of the texture. Optional, defaults to `PIXI.BaseTexture.defaultOptions.scaleMode`. */
-    scaleMode?: SCALE_MODES;
```
Comment (bigtimebuddy) [inline packages/core/src/renderTexture/GenerateTextureSystem.ts:17]: Why did you remove `scaleMode` from these options?

**Review by GoodBoyDigital — APPROVED** [review]

File: packages/core/src/renderTexture/GenerateTextureSystem.ts (hunk 1 above)
Comment (dev7355608) [reply packages/core/src/renderTexture/GenerateTextureSystem.ts:17]: I made `IGenerateTextureOptions` extend `IBaseTextureOptions`.

Comment (bigtimebuddy) [reply packages/core/src/renderTexture/GenerateTextureSystem.ts:17]: Oops missed that!. Thanks for clarifying
